lib/galaxy/datatypes/tabular.py

```diff
@@ -62,15 +63,15 @@ def displayable( self, dataset ):
             return False
 
     def get_chunk(self, trans, dataset, offset=0, ck_size=None):
-        with open(dataset.file_name) as f:
-            f.seek(offset)
-            ck_data = f.read(ck_size or trans.app.config.display_chunk_size)
-            if ck_data and ck_data[-1] != '\n':
+        f = compression_utils.get_fileobj(dataset.file_name)
```

Comment:
I think we can do `with compression_utils.get_fileobj(dataset.file_name) as f:`.